tests/rust/src/bin/stdio.rs

```diff
@@ -1,13 +0,0 @@
-use wasi_tests::{STDERR_FD, STDIN_FD, STDOUT_FD};
-
-unsafe fn test_stdio() {
-    for fd in &[STDIN_FD, STDOUT_FD, STDERR_FD] {
-        wasi::fd_fdstat_get(*fd).expect("fd_fdstat_get on stdio");
-        wasi::fd_renumber(*fd, *fd + 100).expect("renumbering stdio");
```

Comment:
This test looks fine to me; it validates whether `fd_renumber` works on STDIO file descriptors (we have another `renumber.rs` test that operates on file descriptors). As of today, nothing in the documentation says this function should not work on those descriptors; I'd suggest first opening a PR that clarifies the behavior here: https://github.com/WebAssembly/WASI/blob/main/legacy/preview1/witx/wasi_snapshot_preview1.witx if you believe this function should not work on STDIO; unless it's specified, I'd treat stdio as regular file descriptors here (i.e. the test should work).

Reply:
Sorry, I should have been more clear - the issue isn't with the STDIO file descriptors. It's that the destination file descriptor is an arbitrary number i.e. it doesn't refer to any actual open file descriptor.

The [docs](https://github.com/WebAssembly/WASI/blob/main/legacy/preview1/witx/wasi_snapshot_preview1.witx#L245-L251) imply that this behaviour is not supported if I understand correctly.

Although actually, if we still want to validate that renumbering stdio fd's is possible, I can alter the test to renumber the stdio fd's to valid fd's.

Reply:
I've changed the test so that the destination file descriptors are valid.

Reply:
I wonder then whether we should also have a test where we renumber to an arbitrary file descriptor and in that case expect the failure?

Reply:
Yes, that's correct - we should expect failure if the destination is an invalid fd. I'll add a test case in the existing `renumber.rs` test.